packages/database/src/migrations/20231027053345-AddDashboardMailingListTable-modifies-schema.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.createTable('dashboard_mailing_list', {
+    columns: {
+      id: { type: 'text', primaryKey: true },
+      email: { type: 'text', notNull: true },
+      dashboard_id: {
+        type: 'text',
+        notNull: true,
+        foreignKey: {
+          name: `dashboard_mailing_list_dashboard_id_fk`,
+          table: 'dashboard',
+          rules: {
+            onDelete: 'CASCADE',
+            onUpdate: 'CASCADE',
+          },
+          mapping: 'id',
+        },
+      },
+      subscribed: { type: 'boolean', defaultValue: true, notNull: true },
+      unsubscribed_time: { type: 'timestamp' },
+    },
+    ifNotExists: true,
+  });
+  await db.runSql(
+    `ALTER TABLE dashboard_mailing_list ADD CONSTRAINT dashboard_id_email_unique UNIQUE (dashboard_id, email)`,
```

Comment:
Felt this uniqueness constraint is appropriate as we don't want multiple mailing list entries for the same user to the same dashboard. We'll need to add logic in the routes to ensure we first check if the entry exists before creating/editing